dermaknight/app/ui/quizComponents/QuizPage.js

```diff
@@ -0,0 +1,88 @@
+'use client';
+import {useRouter} from 'next/navigation';
+
+
+import styles from './QuizPage.module.css'; 
+import { useState } from 'react';
+import QuizButton from './QuizButton';
+import ProgressBar from './ProgressBar';
+import { quizQuestions } from '@/app/data/quizData';
+
+export default function QuizPage() {
+  const [currentQuestion, setCurrentQuestion] = useState(0);
+  const [activeButton, setActiveButton] = useState(null);
+  const [answers, setAnswers] = useState({});
+
+  const handleButtonClick = (buttonId) => {
+    setActiveButton(buttonId);
+    setAnswers({
+      ...answers,
+      [currentQuestion]: buttonId
+    });
+  };
+
+  const handleNextQuestion = () => {
+    if (currentQuestion < quizQuestions.length - 1) {
+      setCurrentQuestion(currentQuestion + 1);
+      setActiveButton(answers[currentQuestion + 1] || null);
+    }
+  };
+
+  const handlePreviousQuestion = () => {
+    if (currentQuestion > 0) {
+      setCurrentQuestion(currentQuestion - 1);
+      setActiveButton(answers[currentQuestion - 1] || null);
+    }
+  };
+
+  const currentQuestionData = quizQuestions[currentQuestion];
+  const isSkinToneQuestion = currentQuestionData.id === 2;
```

Comment:
Don't love the magic number...leave a comment why this is like this.

Reply:
If you can't find a better way.

Reply:
Maybe add a boolean isSkinToneQuestion to your object(s).

if(isSkinToneQuestion) ....